metricbeat/module/redis/metricset.go

```diff
@@ -39,21 +42,38 @@ func NewMetricSet(base mb.BaseMetricSet) (*MetricSet, error) {
 		IdleTimeout time.Duration `config:"idle_timeout"`
 		Network     string        `config:"network"`
 		MaxConn     int           `config:"maxconn" validate:"min=1"`
-		Password    string        `config:"password"`
 	}{
-		Network:  "tcp",
-		MaxConn:  10,
-		Password: "",
+		Network: "tcp",
+		MaxConn: 10,
 	}
 	err := base.Module().UnpackConfig(&config)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to read configuration")
 	}
 
+	uriParsed, _ := url.Parse(base.HostData().URI)
+	if uriParsed.RawQuery != "" {
+		queries := strings.Split(uriParsed.RawQuery, "&")
```

Comment:
Could you use [URL.Query()](https://golang.org/pkg/net/url/#URL.Query) to parse this?

Reply:
yes 👍 thanks!